java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightInfo.java

```diff
@@ -130,6 +131,14 @@ public FlightInfo(Schema schema, FlightDescriptor descriptor, List<FlightEndpoin
     option = IpcOption.DEFAULT;
   }
 
+  public Optional<Schema> getSchemaOptional() {
+    return Optional.ofNullable(schema);
+  }
+
+  /**
+   * Returns the schema, or null if no schema is present.
```

Comment:
For maximum compatibility, we could return a schema of no fields if the underlying schema is null, the same way it appears to have done before.